backend/src/common/types/service.type.ts

```diff
@@ -4,7 +4,7 @@ type Service<T = unknown> = {
         items: T[];
     }>;
     create(payload: unknown): Promise<T>;
-    update(): Promise<T>;
+    update(id: number, updatedDetails: unknown): Promise<T | null>;
```

Comment:
```suggestion
    update(query: Record<string, T>, payload: unknown): Promise<T | null>;
```
I think better use name **query** and **payload** as it is general type, so in some cases we find item to update not only by id